host/http.go

```diff
@@ -215,6 +227,15 @@ func (h *jobAPI) StopJob(w http.ResponseWriter, r *http.Request, ps httprouter.P
 	w.WriteHeader(200)
 }
 
+func (h *jobAPI) DeregisterJob(w http.ResponseWriter, r *http.Request, ps httprouter.Params) {
```

Comment:
Maybe this should be called `DiscoverdDeregisterJob` or something like that? It's not clear what it is being deregistered from.